### PR TITLE
Remove unnecessary async/await

### PR DESCRIPTION
**Pull request by BjarkeMeier**
## Description

Inspired by the section "Consider using return Task instead of return await" of [this article](https://medium.com/@deep_blue_day/long-story-short-async-await-best-practices-in-net-1f39d7d84050).

## References

* #926

### CODE REVIEW

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/Dtos/ChargeCommands/Validation/BusinessValidation/Factories/ChargeCommandBusinessValidationRulesFactory.cs [hunk 1]
```diff
@@ -97,18 +97,17 @@ public async Task<IValidationRuleSet> CreateRulesAsync(ChargeCommand chargeComma
             return rules;
         }
 
-        private async Task<Charge?> GetChargeOrNullAsync(ChargeCommand command)
+        private Task<Charge?> GetChargeOrNullAsync(ChargeCommand command)
```
Comment (HenrikSommer) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/Dtos/ChargeCommands/Validation/BusinessValidation/Factories/ChargeCommandBusinessValidationRulesFactory.cs:100]:
```suggestion
        private Task<Charge?> GetChargeOrNull(ChargeCommand command)
```

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/Dtos/ChargeCommands/Validation/BusinessValidation/Factories/ChargeCommandBusinessValidationRulesFactory.cs [hunk 2]
```diff
@@ -97,18 +97,17 @@ public async Task<IValidationRuleSet> CreateRulesAsync(ChargeCommand chargeComma
...
         {
             var chargeId = command.ChargeOperation.ChargeId;
             var chargeOperationChargeOwner = command.ChargeOperation.ChargeOwner;
             var chargeType = command.ChargeOperation.Type;
 
-            return await _chargeRepository
+            return _chargeRepository
                 .GetOrNullAsync(new ChargeIdentifier(
```
Comment (HenrikSommer) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/Dtos/ChargeCommands/Validation/BusinessValidation/Factories/ChargeCommandBusinessValidationRulesFactory.cs:107]: It wasn't await'able before?

**Review by HenrikSommer — COMMENTED** [review]
Just curious, what were your criteria for assessing that async/await was unnecessary in these cases?

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/Dtos/ChargeCommands/Validation/BusinessValidation/Factories/ChargeCommandBusinessValidationRulesFactory.cs (hunk 1 above)
Comment (BjarkeMeier) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/Dtos/ChargeCommands/Validation/BusinessValidation/Factories/ChargeCommandBusinessValidationRulesFactory.cs:100]: I don't agree. The nature of the method remains asynchronous. Whether it has the `async` modifier telling the compiler to generate (yet another) state machine or not is an implementation detail. I believe this aligns with the guidelines from Stephen Toub [here](https://stackoverflow.com/questions/15951774/does-the-use-of-the-async-suffix-in-a-method-name-depend-on-whether-the-async) (quoted in the accepted answer).

**Comment by BjarkeMeier** [comment]
> Just curious, what were your criteria for assessing that async/await was unnecessary in these cases?

async/await causes the compiler to create a state machine class to manage the logic of your method. This causes an unnecessary overhead, which can be avoided in some cases. For more details you may want to read section "Consider using return Task instead of return await" of [this article](https://medium.com/@deep_blue_day/long-story-short-async-await-best-practices-in-net-1f39d7d84050).

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/Dtos/ChargeCommands/Validation/BusinessValidation/Factories/ChargeCommandBusinessValidationRulesFactory.cs (hunk 1 above)
Comment (BjarkeMeier) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/Dtos/ChargeCommands/Validation/BusinessValidation/Factories/ChargeCommandBusinessValidationRulesFactory.cs:100]: Btw, we also have this rule actually enforcing the naming convention I advocate:
image

Comment (HenrikSommer) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Domain/Dtos/ChargeCommands/Validation/BusinessValidation/Factories/ChargeCommandBusinessValidationRulesFactory.cs:100]: Oh! You're right, it still returns `Task` :-)

**Review by HenrikSommer — APPROVED** [review]

**Pull request merged**